deps/linenoise/linenoise.c

```diff
@@ -1067,14 +1157,23 @@ static char *linenoiseNoTTY(void) {
  * editing function or uses dummy fgets() so that you will be able to type
  * something even in the most desperate of the conditions. */
 char *linenoise(const char *prompt) {
-    char buf[LINENOISE_MAX_LINE];
+    return linenoiseWithBuffer(prompt, NULL, 0);
+}
+
+char *linenoiseWithBuffer(const char *prompt, const char *initial_buf, const int initial_buf_len) {
+    only_refresh_prompt = 0;
+    char buf[LINENOISE_MAX_LINE] = {0};
+    if (initial_buf_len) {
+        memcpy(buf, initial_buf, initial_buf_len);
```

Comment:
Don't we need to check that initial_buf_len is less than sizeof(buf)?

Reply:
yes! good catch. thanks! will update